bbc-basic/step7_quote.bbc

```diff
@@ -53,25 +53,29 @@ END
 DEF FNREAD(a$)
 =FNread_str(FNalloc_string(a$))
 
-DEF FNis_pair(val%)
-=FNis_seq(val%) AND NOT FNis_empty(val%)
-
-DEF FNquasiquote(ast%)
-  LOCAL car%, caar%
-  IF NOT FNis_pair(ast%) THEN =FNalloc_list2(FNalloc_symbol("quote"), ast%)
-  car% = FNfirst(ast%)
-  IF FNis_symbol(car%) THEN
-    IF FNunbox_symbol(car%) = "unquote" THEN =FNnth(ast%, 1)
-  ENDIF
-  IF FNis_pair(car%) THEN
-    caar% = FNfirst(car%)
-    IF FNis_symbol(caar%) THEN
-      IF FNunbox_symbol(caar%) = "splice-unquote" THEN
-        =FNalloc_list3(FNalloc_symbol("concat"), FNnth(car%, 1), FNquasiquote(FNrest(ast%)))
-      ENDIF
-    ENDIF
+DEF FNstarts_with(ast%, sym$)
+  LOCAL a0%
+  IF NOT FNis_seq(ast%) THEN =FALSE
+  a0% = FNfirst(arg%)
+  =FNis_symbol(a0%) AND FNunbox_symbol(a0%) = sym$
```

Comment:
I think this is unsafe: `AND` in BBC BASIC doesn't short-circuit, so this will call `FNunbox_symbol` even when `a0%` isn't a symbol, which will cause an error. I _think_ it should be possible to test for this by passing suitable input to `quasiquote` but I'm not in a position to experiment with that right now.

I think you need something like this instead:
```
  IF NOT FNis_symbol(a0%) THEN =FALSE
=FNunbox_symbol(a0$) = sym$
```